server/src/main/java/org/elasticsearch/cluster/coordination/CoordinationState.java

```diff
@@ -342,7 +344,7 @@ && getLastCommittedConfiguration().equals(getLastAcceptedConfiguration()) == fal
             throw new CoordinationStateRejectedException("only allow reconfiguration if joinVotes have quorum for new config");
         }
 
-        assert clusterState.getLastCommittedConfiguration().equals(getLastCommittedConfiguration())
+        assert clusterState.getLastCommittedConfiguration().equals(getLastCommittedConfiguration()) || REGISTER_COORDINATION_MODE_ENABLED
```

Comment:
Would you mind if we remove this in a follow-up?

Reply:
I'm a little concerned that this might hide some fundamental issue with the approach we're taking. But I see that it's only weakening this one assertion now which definitely reduces my concern. Maybe you can help me understand why this particular one still needs weakening?

Reply:
We discussed on Zoom; this change relates to the following check which ensures that we do not have overlapping reconfigurations in flight:

https://github.com/elastic/elasticsearch/blob/46b01d9171a17a60ccb305fc609a405c36e147bc/server/src/main/java/org/elasticsearch/cluster/coordination/CoordinationState.java#L337-L341

Stateless masters won't care about this check so we discussed weakening it (i.e. moving it to `ElectionStrategy` so we can skip it) but in fact we don't really care about the voting configuration at all in stateless so we decided to stop doing any reconfigurations and instead just use a fixed dummy voting configuration.

Reply:
Finally I've moved the check to the `ElectionStrategy` we do check in some places that the collected votes form a quorum based on the accepted configuration, so I figured that it would be simpler to move the reconfiguration check to the `ElectionStrategy`

Reply:
Ok, we should probably weaken those checks but we can revisit this in a follow-up.